ark/utils/load_utils.py

```diff
@@ -319,3 +321,197 @@ def load_imgs_from_dir(data_dir, files=None, match_substring=None, trim_suffix=N
+    """Generates the complete tiled fov list when given a list of fov image folders
+
+    Args:
+        folders (list): list of fov folders
+        return_dims (bool): whether to also return row and col dimensions
+    Returns:
+        tuple: names of all fovs expected for tiled image shape, and dimensions if return_dims
+    """
+
+    rows, cols, names, prefix = [], [], {}, False
+
+    # check for run name prefix
+    if folders[0][0] != 'R':
+        prefix = True
+        # dict containing fov name and run name
+        for folder in folders:
+            fov = ''.join(folder.split("_")[-1:])
+            prefix = '_'.join(folder.split("_")[:-1])
+            names[fov] = prefix
+        fov_names = list(names.keys())
+    else:
+        fov_names = folders
+
+    # get tiled image dimensions
+    for fov in fov_names:
+        fov_digits = re.findall(r'\d+', fov)
+        rows.append(int(fov_digits[0]))
+        cols.append(int(fov_digits[1]))
+
+    row_num, col_num = max(rows), max(cols)
+
+    # create list of expected fov names
+    expected_fovs = []
+    for n in range(row_num):
+        for m in range(col_num):
+            dim = f'R{n + 1}C{m + 1}'
+            # prepend run name if had one before
+            if prefix and dim in fov_names:
+                expected_fovs.append(f"{names[dim]}_" + dim)
+            else:
+                expected_fovs.append(dim)
+
+    if return_dims:
+        return expected_fovs, row_num, col_num
+    else:
+        return expected_fovs
+
+
+def get_max_img_size(image_dir, img_sub_folder='', fov_list=None):
+    """Retrieves the maximum FOV image size listed in the run file, or for the given FOVs
+
+    Args:
+        image_dir (str): path to the extracted images for the specific run
+        img_sub_folder (str): optional name of image sub-folder within each fov
+        fov_list (list): list of fovs to check max size for, default none will check all fovs
+    Returns:
+        int: value of max image size
+    """
+
+    img_sizes = []
+
+    # get folder names
+    if not fov_list:
+        fov_list = iou.list_folders(image_dir)
+        if 'stitched_images' in fov_list:
+            fov_list.remove('stitched_images')
+
+    channels = iou.list_files(os.path.join(image_dir, fov_list[0], img_sub_folder))
+
+    # check image size for each fov
+    for fov in fov_list:
+        test_file = io.imread(os.path.join(image_dir, fov, img_sub_folder, channels[0]))
+        img_sizes.append(test_file.shape[1])
+
+    # largest in run
+    max_img_size = max(img_sizes)
+    return max_img_size
+
+
+def load_tiled_img_data(data_dir, img_sub_folder=None, channels=None, max_image_size=None):
+    """Takes a set of (tiled) imgs from a directory structure and loads them into an xarray.
+
+    Args:
+        data_dir (str):
+            directory containing folders of images
+        img_sub_folder (str):
+            optional name of image sub-folder within each fov
+        channels (list):
```

Comment:
Given that this function is only ever passed a single channel, seems like the default behavior here of channels=None doesn't really make sense.